src/gui/pedalboard/cables/CableView.cpp

```diff
@@ -31,45 +31,46 @@ void CableView::paint (Graphics& g)
         }
     }
 
-    g.setColour (cableColour.brighter (0.1f));
-    for (auto* cable : cables)
+    if (cableMouse != nullptr)
     {
-        auto* startEditor = board->findEditorForProcessor (cable->startProc);
-        jassert (startEditor != nullptr);
-
-        auto startPortLocation = CableViewPortLocationHelper::getPortLocation ({ startEditor, cable->startIdx, false });
-
-        if (cable->endProc != nullptr)
+        auto mousePos = cableMouse->getPosition();
+        const auto nearestInputPort = portLocationHelper->getNearestInputPort (mousePos, cables.getLast()->startProc);
+        if (nearestInputPort.editor != nullptr && nearestInputPort.isInput && ! portLocationHelper->isInputPortConnected (nearestInputPort))
         {
-            auto* endEditor = board->findEditorForProcessor (cable->endProc);
-            jassert (endEditor != nullptr);
+            auto endPortLocation = CableViewPortLocationHelper::getPortLocation (nearestInputPort);
+            drawCablePortGlow (g, endPortLocation, scaleFactor);
+        }
+    }
+}
 
-            auto endPortLocation = CableViewPortLocationHelper::getPortLocation ({ endEditor, cable->endIdx, true });
+void CableView::resized()
+{
+    for (auto* cable : cables)
+    {
+        cable->setBounds (getLocalBounds());
+    }
+}
 
-            auto startColour = startEditor->getColour();
-            auto endColour = endEditor->getColour();
-            auto levelDB = cable->endProc->getInputLevelDB (cable->endIdx);
-            drawCable (g, startPortLocation.toFloat(), startColour, endPortLocation.toFloat(), endColour, scaleFactor, levelDB);
-        }
-        else if (connectionHelper->cableMouse != nullptr)
+void CableView::mouseDown (const MouseEvent& e)
+{
+    if (e.eventComponent == nullptr)
+        return; // not a valid mouse event
+
+    if (e.mods.isPopupMenu())
+    {
+        for (auto* cable : cables)
         {
-            auto mousePos = connectionHelper->cableMouse->getPosition();
-            const auto nearestInputPort = portLocationHelper->getNearestInputPort (mousePos, cable->startProc);
-            if (nearestInputPort.editor != nullptr && nearestInputPort.isInput && ! portLocationHelper->isInputPortConnected (nearestInputPort))
+            juce::Point clicked (e.getMouseDownX(), e.getMouseDownY());
+            if (cable->contains (clicked)) // Checks hitTest method for cable
             {
-                auto endPortLocation = CableViewPortLocationHelper::getPortLocation (nearestInputPort);
```

Comment:
Oh interesting, thanks for catching that. I think ideally this would be fixed, even testing changes today, I went to delete a cable and was annoyed when I couldn't click the port when my mouse was over a cable.

I experimented with setting the component in `getNearestPort()` to `board` and that seems to fix the issue. I won't push this change yet as I'm not sure what the side effect of always having that component set to `board` could be, but let me know if you think this could work, or if we should look into a better solution.

![Screen Shot 2022-08-10 at 5 41 03 PM](https://user-images.githubusercontent.com/43660096/184041300-cab2f8a2-eb2e-4eb5-8c5d-be14a89bc265.png)

Reply:
Yeah, I think that should work... with that change do the ports still glow when the mouse is hovering over the part of the port which is on the module editor? I worry that might get skipped if always using the `board` as the reference component.

Reply:
Yup, they are still glowing using the board. I implemented this change in the newest set of changes.